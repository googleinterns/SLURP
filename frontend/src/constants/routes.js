/**
 * This file specifies the pages that a user can visit.
 */
export const LANDING = '/';
export const SIGN_IN = '/signin';
export const VIEW_TRIPS = '/view-trips';
export const VIEW_ACTIVITIES = '/view-activities';
