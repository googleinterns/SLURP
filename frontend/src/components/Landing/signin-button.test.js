import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import renderer from 'react-test-renderer';

import SignInButton from './signin-button.js';
import { SIGN_IN } from '../../constants/routes.js';

// Mock the push function from react-router-dom's useHistory.
const mockHistoryPush = jest.fn();
jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    push: mockHistoryPush
  })
}));

describe('SignInButton component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<SignInButton />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('redirects to signin page on click', () => {
    const { getByText } = render(<SignInButton />);
    fireEvent.click(getByText(/Go/));
    expect(mockHistoryPush).toBeCalledWith(SIGN_IN);
  })
});
