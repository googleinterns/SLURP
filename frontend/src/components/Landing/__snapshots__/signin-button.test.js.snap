// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInButton component renders correctly 1`] = `
<button
  className="btn btn-primary btn-lg"
  disabled={false}
  onClick={[Function]}
  type="button"
>
  Sign In
</button>
`;
